js/controllers/modal/ModalHandler.js

```diff
@@ -47,7 +47,7 @@
 				titleIcon: string }} */ (this.mergeOptions(internalOptions));
 
 		// Attach content to the modal
-		$handledElement.html(this.modalBuild().html());
+		$handledElement.html(this.modalBuild()[0].outerHTML);
```

Comment:
You probably saw this, but I kept my tweaks as separate commits from the squashed one containing your work. That'll hopefully make it easier to track down anything I inadvertently introduced.